### PR TITLE
Merge from wnprc18.3 svn r.64261 to r.65647

### PR DESCRIPTION
**Pull request by labkey-bpatel**
#### Rationale
Merge from wnprc18.3 svn r.64261 to r.65647 (skipped r. 64864 & 65166 since these were backports)

#### Related Pull Requests
LabKey/commonAssays#200
https://github.com/LabKey/LabDevKitModules/pull/38
LabKey/onprcEHRModules#20
LabKey/platform#1370
LabKey/tnprc_billing#72

### CODE REVIEW

File: ehr/resources/web/ehr/model/sources/Necropsy.js [hunk 1]
```diff
@@ -1,35 +0,0 @@
-/*
```
Comment (labkey-jeckels) [inline ehr/resources/web/ehr/model/sources/Necropsy.js]: This file is still used in ONPRC and SNPRC, so we can't delete it

File: ehr/resources/web/ehr/model/sources/Biopsy.js [hunk 2]
```diff
@@ -1,64 +0,0 @@
-/*
```
Comment (labkey-jeckels) [inline ehr/resources/web/ehr/model/sources/Biopsy.js]: This file is still used in ONPRC and SNPRC, so we can't delete it

File: ehr/resources/web/ehr/model/sources/Surgery.js [hunk 3]
```diff
@@ -1,117 +0,0 @@
-/*
```
Comment (labkey-jeckels) [inline ehr/resources/web/ehr/model/sources/Surgery.js]: This file is still used in ONPRC and SNPRC, so we can't delete it

File: ehr/src/org/labkey/ehr/query/EHRCustomPermissionsTable.java [hunk 4]
```diff
@@ -49,7 +34,6 @@ protected Container getDomainContainer()
         return EHRDomainKind.getDomainContainer(getContainer());
     }
 
-    @Override
```
Comment (labkey-jeckels) [inline ehr/src/org/labkey/ehr/query/EHRCustomPermissionsTable.java]: Why are these lines deleted? Doesn't seem to be a useful change

File: ehr/src/org/labkey/ehr/query/EHRUserSchema.java [hunk 5]
```diff
@@ -24,6 +24,7 @@
 import org.labkey.api.ehr.security.EHRDataAdminPermission;
 import org.labkey.api.ehr.security.EHRProjectEditPermission;
 import org.labkey.api.ehr.security.EHRProtocolEditPermission;
+import org.labkey.api.ehr.security.EHR_BillingAdminPermission;
```
Comment (labkey-jeckels) [inline ehr/src/org/labkey/ehr/query/EHRUserSchema.java]: Why is this added? Doesn't seem to be used

File: ehr/src/org/labkey/ehr/security/EHRRequestorRole.java [hunk 6]
```diff
@@ -26,6 +26,7 @@
 import org.labkey.api.ehr.security.EHRRequestPermission;
 import org.labkey.api.ehr.security.EHRRequestSampleDeliveredInsertPermission;
 import org.labkey.api.ehr.security.EHRRequestSampleDeliveredUpdatePermission;
+import org.labkey.api.ehr.security.EHRScheduledInsertPermission;
```
Comment (labkey-jeckels) [inline ehr/src/org/labkey/ehr/security/EHRRequestorRole.java]: Unused import

File: EHR_ComplianceDB/resources/views/SOPadmin.html [hunk 7]
```diff
@@ -3,6 +3,7 @@
 Ext4.onReady(function(){
     var webpart = <%=webpartContext%>;
     var container = LABKEY.getModuleProperty('ehr_compliancedb', 'EmployeeContainer');
+    var pdfs= LABKEY.getModuleProperty('ehr_compliancedb', 'PDFContainer')
```
Comment (labkey-jeckels) [inline EHR_ComplianceDB/resources/views/SOPadmin.html]: Not required for merge, but this and the corresponding .view.xml and .webpart.xml files should be in WNPRC's module, since they're hard-coded to use /WNPRC and don't seem to be used anywhere else.

File: EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/All SOPs.qview.xml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 <customView xmlns="http://labkey.org/data/xml/queryCustomView">
     <columns>
         <column name="SOP_ID"/>
-        <column name="SOP_ID/Name">
+         <column name="Name">
```
Comment (labkey-jeckels) [inline EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/All SOPs.qview.xml]: This file should move to WNPRC_EHR since the SOPRequirements moved

File: EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read Standard.qview.xml [hunk 9]
```diff
@@ -1,7 +1,7 @@
 <customView xmlns="http://labkey.org/data/xml/queryCustomView">
     <columns>
         <column name="SOP_ID"/>
-        <column name="SOP_ID/Name">
+        <column name="Name">
```
Comment (labkey-jeckels) [inline EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read Standard.qview.xml]: This file should move to WNPRC_EHR since the SOPRequirements moved

File: EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read.qview.xml [hunk 10]
```diff
@@ -1,12 +1,12 @@
 <customView xmlns="http://labkey.org/data/xml/queryCustomView">
     <columns>
         <column name="SOP_ID"/>
-        <column name="SOP_ID/Name">
+        <column name="Name">
```
Comment (labkey-jeckels) [inline EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read.qview.xml]: This file should move to WNPRC_EHR since the SOPRequirements moved

File: Viral_Load_Assay/resources/web/Viral_Load_Assay/SampleData/LC_data.txt [hunk 11]
```diff
@@ -1,28 +0,0 @@
-TRUE	1	STD_15000000	15.19	1.47E+07	15000000
```
Comment (labkey-jeckels) [inline Viral_Load_Assay/resources/web/Viral_Load_Assay/SampleData/LC_data.txt:1]: We should also delete Viral_Load_Assay\src\org\labkey\viral_load_assay\assay\LightCyclerImportMethod.java as it's no longer referenced, and contains a reference to this now deleted file

File: ehr/resources/web/ehr/grid/Panel.js [hunk 12]
```diff
@@ -49,19 +49,13 @@ Ext4.define('EHR.grid.Panel', {
         this.addEvents('animalchange', 'storevalidationcomplete');
         this.enableBubble('animalchange');
 
-        this.getSelectionModel().on('selectionchange', function(sm, models){
-            if (models.length != 1)
-                return;
+        this.getSelectionModel().on('selectionchange', this.handleSectionChangeEvent, this);
```
Comment (labkey-jeckels) [inline ehr/resources/web/ehr/grid/Panel.js:52]: Looks like a bad merge - the lines below about byPassId should be inside the handleSelectionChangeEvent function below

File: ehr/resources/web/ehr/form/field/ProjectEntryField.js [hunk 13]
```diff
@@ -10,351 +10,352 @@
...
-    alias: 'widget.ehr-projectentryfield',
-
-    fieldLabel: 'Project',
-    typeAhead: true,
-    forceSelection: true, //NOTE: has been re-enabled, but it is important to let the field get set prior to loading
-    emptyText:'',
-    disabled: false,
-    matchFieldWidth: false,
-    includeDefaultProjects: true,
-    invesLastNameCol: 'lastName',
```
Comment (labkey-jeckels) [inline ehr/resources/web/ehr/form/field/ProjectEntryField.js:23]: This looks like a problem, as TNPRC is overriding this value and this change appears to completely snip out the investigator value from the template. We should coordinate with WNPRC on the motivation for this change.

File: ehr/src/org/labkey/ehr/EHRController.java [hunk 14]
```diff
@@ -375,8 +376,32 @@ public ModelAndView getView(EHRQueryForm form, BindException errors)
                         url.addParameter("importURL", importUrl.toString());
                     }
                 }
-                else
+                //TODO implement this after view is set up for editable react form
+                else if (isReactForm(form.getSchemaName(), form.getQueryName()))
                 {
+                    //weight is custom since it was introduced in its own controller
+                    if ("weight".equals(form.getQueryName())){
```
Comment (labkey-jeckels) [inline ehr/src/org/labkey/ehr/EHRController.java]: We should find an approach that doesn't require hard-coding of WNPRC in general, or of the oddities of the weights form in particular. Could isReactForm() return the URL to use based on the values in the TABLE_FORM_FRAMEWORK_TYPES table?

File: ehr_billing/src/org/labkey/ehr_billing/notification/BillingNotification.java [hunk 15]
```diff
@@ -342,7 +342,7 @@ public void exec(ResultSet object) throws SQLException
                     projectDisplay = "None";
                 }
 
-                String financialAnalyst = rs.getString(FieldKey.fromString("investigatorLastName"));
+                String financialAnalyst = rs.getString(FieldKey.fromString("investigator"));
```
Comment (labkey-jeckels) [inline ehr_billing/src/org/labkey/ehr_billing/notification/BillingNotification.java:345]: @labkey-bpatel  and @labkey-ankurj can you confirm this is OK for both WNPRC and TNPRC?

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]

File: ehr_billing/src/org/labkey/ehr_billing/notification/BillingNotification.java (hunk 15 above)
Comment (labkey-ankurj) [reply ehr_billing/src/org/labkey/ehr_billing/notification/BillingNotification.java:345]: I am not sure if I've come across/tested runNotification functionality in TNPRC, but a text search for investigatorLastName in both of tnprc repositories has no results.

Comment (labkey-ankurj) [reply ehr_billing/src/org/labkey/ehr_billing/notification/BillingNotification.java:345]: Looks like there is RequestAdminNotification in tnprc_ehr and ehrModules/EnterDataPanel.js has the key `org.labkey.onprc_ehr.notification.RequestAdminNotification`

File: ehr/resources/web/ehr/model/sources/Necropsy.js (hunk 1 above)
Comment (labkey-bpatel) [reply ehr/resources/web/ehr/model/sources/Necropsy.js]: Thanks for double checking, now restored.

File: ehr/resources/web/ehr/model/sources/Biopsy.js (hunk 2 above)
Comment (labkey-bpatel) [reply ehr/resources/web/ehr/model/sources/Biopsy.js]: Restored

File: ehr/resources/web/ehr/model/sources/Surgery.js (hunk 3 above)
Comment (labkey-bpatel) [reply ehr/resources/web/ehr/model/sources/Surgery.js]: Restored

File: ehr/src/org/labkey/ehr/query/EHRCustomPermissionsTable.java (hunk 4 above)
Comment (labkey-bpatel) [reply ehr/src/org/labkey/ehr/query/EHRCustomPermissionsTable.java]: Not sure, looks like it took what was in wnprc18.3 branch. Added back in.

File: ehr/src/org/labkey/ehr/query/EHRUserSchema.java (hunk 5 above)
Comment (labkey-bpatel) [reply ehr/src/org/labkey/ehr/query/EHRUserSchema.java]: removed

File: ehr/src/org/labkey/ehr/security/EHRRequestorRole.java (hunk 6 above)
Comment (labkey-bpatel) [reply ehr/src/org/labkey/ehr/security/EHRRequestorRole.java]: removed

File: Viral_Load_Assay/resources/web/Viral_Load_Assay/SampleData/LC_data.txt (hunk 11 above)
Comment (labkey-bpatel) [reply Viral_Load_Assay/resources/web/Viral_Load_Assay/SampleData/LC_data.txt:1]: deleted

File: ehr/resources/web/ehr/grid/Panel.js (hunk 12 above)
Comment (labkey-bpatel) [reply ehr/resources/web/ehr/grid/Panel.js:52]: good eye! thank you! now fixed

File: ehr/resources/web/ehr/form/field/ProjectEntryField.js (hunk 13 above)
Comment (labkey-bpatel) [reply ehr/resources/web/ehr/form/field/ProjectEntryField.js:23]: I've restored invesLastNameCol for now, will sync with WNPRC

File: ehr_billing/src/org/labkey/ehr_billing/notification/BillingNotification.java (hunk 15 above)
Comment (labkey-bpatel) [reply ehr_billing/src/org/labkey/ehr_billing/notification/BillingNotification.java:345]: WNPRC has been using this in production, so looks like that change is fine.
Thanks @labkey-ankurj for checking in tnprc.

File: EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/All SOPs.qview.xml (hunk 8 above)
Comment (labkey-bpatel) [reply EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/All SOPs.qview.xml]: Removed from ehr_compliancedb module, and added to wnprc-modules: https://github.com/WNPRC-EHR-Services/wnprc-modules/tree/20.6_fb_wnprc183_192Merge

File: EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read Standard.qview.xml (hunk 9 above)
Comment (labkey-bpatel) [reply EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read Standard.qview.xml]: Removed from ehr_compliancedb module, and added to wnprc-modules: https://github.com/WNPRC-EHR-Services/wnprc-modules/tree/20.6_fb_wnprc183_192Merge

File: EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read.qview.xml (hunk 10 above)
Comment (labkey-bpatel) [reply EHR_ComplianceDB/resources/queries/ehr_compliancedb/SOPrequirements/To Read.qview.xml]: Removed from ehr_compliancedb module, and added to wnprc-modules: https://github.com/WNPRC-EHR-Services/wnprc-modules/tree/20.6_fb_wnprc183_192Merge

File: EHR_ComplianceDB/resources/views/SOPadmin.html (hunk 7 above)
Comment (labkey-bpatel) [reply EHR_ComplianceDB/resources/views/SOPadmin.html]: done! Removed from ehr_compliancedb module, and added to wnprc-modules: https://github.com/WNPRC-EHR-Services/wnprc-modules/tree/20.6_fb_wnprc183_192Merge

File: ehr/resources/web/ehr/form/field/ProjectEntryField.js (hunk 13 above)
Comment (labkey-bpatel) [reply ehr/resources/web/ehr/form/field/ProjectEntryField.js:23]: Email sent to client about invesLastNameCol.

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]
This looks good, pending Chad's revisions to EHRController to avoid hard-coding the React form URLs

**Review by labkey-jeckels — APPROVED** [review]
Looks good, pending TeamCity happiness
